Ethiopia-COVID19/bot
# Add FB Messenger example

I'm creating another server example with a FB Messenger bot webhook endpoint. 

Now, the old telegram app can be found in `/telegram`. `./messenger` has a simple Node application that exposes a webhook endpoint for the FB Messenger Platform.

**How do I test this end to end?**
The messenger platform needs to have access to your webhook endpoint on the public web over https. The fastest way to test things, if you are running the server locally, is to expose your 5000 port using ngrok.

0. Update the `messeger\config\default.json` with your FB Application secrets from their [getting started guide](https://developers.facebook.com/docs/messenger-platform/getting-started/app-setup).

1. Start the node application on port 5000
```
node app.js
```

2. Sanity check that the application is up by opening `http://localhost:5000` in a browser

3. In a separate terminal run:
```
ngrok http 5000
```

NOTE:
In the long run, these should really be one server with different endpoints for each bot telegram, messenger, etc.
